src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -110,44 +144,51 @@ get_graph_id(const pt::ptree& aie_meta, const std::string& graph_name)
-    rtp.selector_lock_id = rtp_node.second.get<uint16_t>("selector_lock_id");
-    rtp.selector_addr = rtp_node.second.get<uint64_t>("selector_address");
-
-    rtp.ping_row = rtp_node.second.get<uint16_t>("ping_buffer_row");
-    rtp.ping_col = rtp_node.second.get<uint16_t>("ping_buffer_column");
-    rtp.ping_lock_id = rtp_node.second.get<uint16_t>("ping_buffer_lock_id");
-    rtp.ping_addr = rtp_node.second.get<uint64_t>("ping_buffer_address");
-
-    rtp.pong_row = rtp_node.second.get<uint16_t>("pong_buffer_row");
-    rtp.pong_col = rtp_node.second.get<uint16_t>("pong_buffer_column");
-    rtp.pong_lock_id = rtp_node.second.get<uint16_t>("pong_buffer_lock_id");
-    rtp.pong_addr = rtp_node.second.get<uint64_t>("pong_buffer_address");
-
-    rtp.is_plrtp = rtp_node.second.get<bool>("is_PL_RTP");
-    rtp.is_input = rtp_node.second.get<bool>("is_input");
-    rtp.is_async = rtp_node.second.get<bool>("is_asynchronous");
-    rtp.is_connected = rtp_node.second.get<bool>("is_connected");
-    rtp.require_lock = rtp_node.second.get<bool>("requires_lock");
-
-    rtps.emplace_back(std::move(rtp));
+    if (rtp_node.second.get<int>("graph_id") != graph_id)
+      continue;
+
+    adf::rtp_config rtp;
+    rtp.portId = rtp_node.second.get<int>("port_id");
+    rtp.aliasId = rtp_node.second.get<int>("alias_id");
+    rtp.portName = rtp_node.second.get<std::string>("port_name");
+    rtp.aliasName = rtp_node.second.get<std::string>("alias_name");
+    rtp.graphId = rtp_node.second.get<int>("graph_id");
+    rtp.numBytes = rtp_node.second.get<size_t>("number_of_bytes");
+    rtp.selectorRow = rtp_node.second.get<short>("selector_row");
+    rtp.selectorColumn = rtp_node.second.get<short>("selector_column");
+    rtp.selectorLockId = rtp_node.second.get<unsigned short>("selector_lock_id");
+    rtp.selectorAddr = rtp_node.second.get<size_t>("selector_address");
+
+    rtp.pingRow = rtp_node.second.get<short>("ping_buffer_row");
+    rtp.pingColumn = rtp_node.second.get<short>("ping_buffer_column");
+    rtp.pingLockId = rtp_node.second.get<unsigned short>("ping_buffer_lock_id");
+    rtp.pingAddr = rtp_node.second.get<size_t>("ping_buffer_address");
+
+    rtp.pongRow = rtp_node.second.get<short>("pong_buffer_row");
+    rtp.pongColumn = rtp_node.second.get<short>("pong_buffer_column");
+    rtp.pongLockId = rtp_node.second.get<unsigned short>("pong_buffer_lock_id");
+    rtp.pongAddr = rtp_node.second.get<size_t>("pong_buffer_address");
+
+    rtp.isPL = rtp_node.second.get<bool>("is_PL_RTP");
+    rtp.isInput = rtp_node.second.get<bool>("is_input");
+    rtp.isAsync = rtp_node.second.get<bool>("is_asynchronous");
+    rtp.isConnect = rtp_node.second.get<bool>("is_connected");
+    rtp.hasLock = rtp_node.second.get<bool>("requires_lock");
+
+    rtps[rtp.portName] = rtp;
   }
 
   return rtps;
 }
 
 std::vector<gmio_type>
-get_gmio(const pt::ptree& aie_meta)
+get_old_gmio(const pt::ptree& aie_meta)
```

Comment:
Can you please clarify why we still need a get_old_gmios? Can we replace them with the new gmio metadata container? (gmio_configs)

Reply:
This will be cleaned up in the next pull request that provides the common layer profiling api